src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -155,7 +156,7 @@ else if (SUPPORT_CONTEXT_STRATEGY_CHECKS.contains(checkShortName)) {
             final Set<Integer> childAstLineNoList = getChildAstLineNo(eventAst);
             final int min = Collections.min(childAstLineNoList);
             final int max = Collections.max(childAstLineNoList);
-            for (int currentLine = min; currentLine <= max; currentLine++) {
+            for (int currentLine = min; currentLine <= max + 1; currentLine++) {
```

Comment:
this change is for a special case that if the deletion line is the last line, then although we consider deletion line information in #162, it still can not be matched, because the last line is out of range of current java version. for example:

```java
case 7:  // violation context line 20
     i++; // line 21
    break; // line 22
```
if we delete line 22, and we consider deletion line information 22, it still not ok, because now `getChildAstLineNo` just get 20-21, no 22

Reply:
1. Please check what are the consequences of such a change for INSERT/REPLACE lines as this is a common logic for all types of edits?
2. Can we do it in an opposite way: when there is a DELETE edit take it's line range with the last line - 1 ? So, the above logic will stay as it was without changes.

Reply:
> Please check what are the consequences of such a change for INSERT/REPLACE lines as this is a common logic for all > types of edits?

yes, only the for DELETE 

> Can we do it in an opposite way: when there is a DELETE edit take it's line range with the last line - 1 ? So, the above 
> logic will stay as it was without changes.

yes, I think it is ok. and try it.